lib/dndsections.sty

```diff
@@ -1,95 +1,226 @@
+\titleformat { \section }
+  { \dnd@TitleFont \LARGE \color { titlered } } % format
+  { \thesection \quad } % label
+  { 0pt } % sep
+  { \RaggedRight \setstretch{.8} \nopagebreak  } % before-code
+  [ \smallskip ] % after-code
+
+\titlespacing* { \section }
+  { 0pt } % left
+  { 1.5ex plus .2ex minus .2ex } % before-sep
+  { 0pt } % after-sep
 
 % Subsection
 \titleformat{\subsection}
-{\color{titlered}\dnd@TitleFont\Large\RaggedRight}{\thesubsection\quad}{0pt}{\nopagebreak}
-[\titleline{\color{titlegold}\titlerule[1pt]}]
+  { \dnd@TitleFont \Large \color { titlered } } % format
+  { \thesubsection \quad } % label
+  { 0pt } % sep
+  { \RaggedRight \setstretch{.8} \nopagebreak } % before-code
+  [ \vspace { -.3ex } \titleline { \color { titlegold } \titlerule [ 1pt ] } ] % after-code
+
+\titlespacing*{ \subsection }
+  { 0pt } % left
+  { 1.6ex } % before-sep
+  { .75ex } % after-sep
 
 % Subsubsection
 \titleformat{\subsubsection}
-{\color{titlered}\dnd@TitleFont\large\RaggedRight}{\thesubsubsection\quad}{0pt}{\nopagebreak}[]
+  { \dnd@TitleFont \large \color { titlered } } % format
+  { \thesubsubsection \quad } % label
+  { 0pt } % sep
+  { \RaggedRight \setstretch{.8} \nopagebreak } % before-code
+
+\titlespacing* { \subsubsection }
+  { 0pt } % left
+  { 1.7ex plus .2ex minus .2ex } % before-sep
+  { 0pt } % after-sep
 
 % Paragraph
 \titleformat{\paragraph}[runin]
-{\normalfont\normalsize\bfseries\slshape}{\theparagraph\quad}{0pt}{}[.]
+  { \normalfont \normalsize \bfseries \slshape } % format
+  { \theparagraph \quad } % label
+  { 0pt } % sep
+  {} % before-code
+  [.] % after-code
+
 \titlespacing*{\paragraph}
-{0pt}{\parskip}{\wordsep}
+  { 0pt } % left
+  { \parskip } % before-sep
+  { \wordsep } % after-sep
 
 % Subparagraph
-\titleformat{\subparagraph}[runin]
-{\normalfont\normalsize\bfseries\slshape}{\thesubparagraph\quad}{0pt}{}[.]
-\titlespacing*{\subparagraph}
-{\parindent}{\parskip}{\wordsep}
+\titleformat { \subparagraph } [ runin ]
+  { \normalfont \normalsize \bfseries \slshape } % format
+  { \thesubparagraph \quad } % label
+  { 0pt } % sep
+  {} % before-code
+  [.] % after-code
+
+\titlespacing* { \subparagraph }
+  { \parindent } % left
+  { \parskip } % before-sep
+  { \wordsep } % after-sep
 
 \fi % \ifpkglayout
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 % Special command for magic items, traps, and the like.
-\newcommand{\subtitlesection}[2]{
-	\subsubsection{#1}\vspace{-1ex}
-	\textit{#2}\vspace{1ex}\par
-	} 
-
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\NewDocumentCommand { \DndItemHeader } { m m }
+  {
+    \subsubsection { #1 }
+    \textit { #2 }
+    \par \smallskip \noindent \ignorespaces
+  }
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Special command for magic items, traps, and the like.
```

Comment:
This should probably read `Special command for spells.`

Reply:
Thanks! Fixed.